tests/authentication/04-check-keys-remote.t

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Checks remote ZMQ keys are created and deleted on shutdown.
+. "$(dirname "$0")/test_header"
+set_test_remote_host
+set_test_number 4
+init_suite "${TEST_NAME_BASE}" <<'__SUITE_RC__'
+[cylc]
+[scheduling]
+    [[graph]]
+        R1 = holder
```

Comment:
```suggestion
        R1 = holder => held
```

Argh, sorry @datamel my previous explanation wasn't very clear - by "the `sleep 10` isn't needed" (in the `held` task) I just meant that the task doesn't need any defined `script` defined because it is never going to run. But for the remote test, you do need `held` to be present so that it can be "held" as a waiting task, in order to stop the suite shutting down as soon as the `holder` task finishes. (Because a held suite will still shut down if it is finished - i.e. if there are no waiting tasks present that are being held back from submitting...)

Reply:
(The "Argh" isn't directed at you, btw!)

Reply:
Thanks!

Reply:
@datamel - sorry if I've just confused you at this point (or maybe you just haven't addressed my comment yet in spite of "resolving" the conversation?) - but as currently written the suite will shut down immediately because the hold has no effect in the absence of any waiting tasks, not leaving enough time for the file check (unless you're very lucky with the timing).

Reply:
Very strange, I suspect this has to do with me committing your suggestion on the site and then making changes and rebasing on my local copy, without pulling down the changes first- rookie error soz! Just fixing now.

Reply:
Ah, makes sense - I've only done the same about a million times!